src/app/services/transmart-services/transmart-external-job-resource.service.ts

```diff
@@ -0,0 +1,260 @@
+import {ErrorHelper} from '../../utilities/error-helper';
+import {SubjectSetConstraint} from '../../models/constraint-models/subject-set-constraint';
+import {CombinationConstraint} from '../../models/constraint-models/combination-constraint';
+import {ConstraintMark} from '../../models/constraint-models/constraint-mark';
+import {catchError, map} from 'rxjs/operators';
+import {ExportDataType} from '../../models/export-models/export-data-type';
+import {ExportFileFormat} from '../../models/export-models/export-file-format';
+import {TransmartExternalJob} from '../../models/transmart-models/transmart-external-job';
+
+
+@Injectable()
+export class TransmartExternalJobResourceService {
+
+  /* the export data view, either 'dataTable', 'surveyTable' or 'customFormat'.
+   * The latter is expected here.
+   */
+  private _exportDataView: string;
+  private _customExportJobName: string;
+  private _endpointUrl: string;
+
+  /*
+   * Flag indicating if the subject selection
+    savePatientSet(arg0: any, arg1: any): any {
+        throw new Error("Method not implemented.");
+    } of step 1 should be automatically
+   * saved as subject set in the backend. If true, that subject set is used as the subject constraint
+   * for step 2.
+   */
+  private _autosaveSubjectSets: boolean;
+  private _subjectSetConstraint: SubjectSetConstraint;
+
+  constructor(private appConfig: AppConfig,
+              private http: HttpClient) {
+    this.exportDataView = appConfig.getConfig('export-data-view');
+    this.customExportJobName = appConfig.getConfig('custom-export-job-name');
+    this.endpointUrl = appConfig.getConfig('export-service-url');
+    this.subjectSetConstraint = new SubjectSetConstraint();
+  }
+
+  get subjectSetConstraint(): SubjectSetConstraint {
+    return this._subjectSetConstraint;
+  }
+
+  set subjectSetConstraint(value: SubjectSetConstraint) {
+    this._subjectSetConstraint = value;
+  }
+
+  get exportDataView(): string {
+    return this._exportDataView;
+  }
+
+  set exportDataView(value: string) {
+    this._exportDataView = value;
+  }
+
+  get customExportJobName(): string {
+    return this._customExportJobName;
+  }
+
+  set customExportJobName(value: string) {
+    this._customExportJobName = value;
+  }
+
+  get endpointUrl(): string {
+    return this._endpointUrl;
+  }
+
+  set endpointUrl(value: string) {
+    this._endpointUrl = value;
+  }
+
+  get autosaveSubjectSets(): boolean {
+    return this._autosaveSubjectSets;
+  }
+
+  set autosaveSubjectSets(value: boolean) {
+    this._autosaveSubjectSets = value;
+  }
+
+  // ------------------------------------- export job calls -------------------------------------
+
+  /**
+   * Make a post http request
+   * @param urlPart - the part used in baseUrl/urlPart
+   * @param body
+   * @param responseField
+   * @returns {Observable<any | any>}
+   */
+  private postCall(urlPart, body, responseField) {
```

Comment:
Such code duplication can be prevented by moving these functions to a separate helper class.

Reply:
Or have a common superclass.